### PR TITLE
Add Sanskrit Shlokas Dataset

### PR DESCRIPTION
**Pull request by bp-high**
## Checklist for adding MMTEB dataset
Reason for dataset addition:
- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [X] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [X] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        dialect=[],
+        text_creation="found",
+        bibtex_citation=None,
+        n_samples={"train": 383, "test": 96},
+        avg_character_length={"train":276 , "test":170 },
+    )
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.rename_column("Sloka", "text")
+        self.dataset = self.dataset.rename_column("Class", "label")
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/san/SanskritShlokasClassification.py:37]:
```suggestion
        self.dataset = self.dataset.rename_columns({"Sloka": "text", "Class": "label"})
```

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/san/SanskritShlokasClassification.py:1]: Please run `make lint` or `make pr` and push changes

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        description="This data set contains ~500 Shlokas  ",
+        reference="https://github.com/goru001/nlp-for-sanskrit",
+        dataset={
+            "path": "bpHigh/iNLTK_Sanskrit_Shlokas_Dataset",
+            "revision": "772c6ef188a4ffd2bd2e115fcefe5957f2a1dc1f",
+        },
+        type="Classification",
+        category="s2s",
+        date=("2014-01-01", "2018-01-01"),
+        eval_splits=["test"],
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/san/SanskritShlokasClassification.py:19]: There is no test split on HF, the only available splits are `train` and `validation`
```suggestion
        eval_splits=["train", "validation"],
```

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        form=["written"],
+        domains=["Religious Text"],
+        task_subtypes=["Topic classification"],
+        license="CC BY-SA 4.0",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation=None,
+        n_samples={"train": 383, "test": 96},
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/san/SanskritShlokasClassification.py:31]:
```suggestion
        n_samples={"train": 383, "validation": 96},
```

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        avg_character_length={"train":276 , "test":170 },
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/san/SanskritShlokasClassification.py:32]:
```suggestion
        avg_character_length={"train":276 , "validation":170 },
```

**Review by imenelydiaker — COMMENTED** [review]
Thank you for this additon! I made some comments. 
Alos, would you please run `intfloat/multilingual-e5-small` and `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2` on this task and push results, thank you.

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py (hunk 1 above)
Comment (bp-high) [reply mteb/tasks/Classification/san/SanskritShlokasClassification.py]: Done

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py (hunk 3 above)
Comment (bp-high) [reply mteb/tasks/Classification/san/SanskritShlokasClassification.py]: Done

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py (hunk 5 above)
Comment (bp-high) [reply mteb/tasks/Classification/san/SanskritShlokasClassification.py]: Done

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py (hunk 2 above)
Comment (bp-high) [reply mteb/tasks/Classification/san/SanskritShlokasClassification.py:1]: Done

File: mteb/tasks/Classification/san/SanskritShlokasClassification.py (hunk 4 above)
Comment (bp-high) [reply mteb/tasks/Classification/san/SanskritShlokasClassification.py:51]: Done

**Comment by bp-high** [comment]
@imenelydiaker have added the requested changes.

**Review by imenelydiaker — COMMENTED** [review]
LGTM! Please add points and we'll be ready to merge 🙂

**Review by imenelydiaker — APPROVED** [review]

**Comment by bp-high** [comment]
Hi @imenelydiaker can we rerun the tests. I think we are getting an unrelated test failure which has been resolved now: #566

**Comment by imenelydiaker** [comment]
@bp-high can you please merge main into your branch?

**Comment by bp-high** [comment]
> @bp-high can you please merge the main branch of this repo into your branch?

Done

**Pull request merged**